Classes/Controller/AuthenticationController.php

```diff
@@ -70,17 +68,17 @@ public function connectAction()
                 'expected' => $_SESSION['oidc_state'],
                 'actual' => $_GET['state'],
             ]);
-            if (!(bool)$this->globalSettings['oidcDisableCSRFProtection']) {
+            if (!(bool) $this->globalSettings['oidcDisableCSRFProtection']) {
                 throw new \RuntimeException('Invalid state', 1489658206);
             }
             static::getLogger()->warning('Bypassing CSRF attack mitigation protection according to the extension configuration');
         }
 
         $loginUrl = $_SESSION['oidc_login_url'];
-        $loginUrl .= strpos($loginUrl, '?') !== false ? '&' : '?';
-        $loginUrl .= 'logintype=login&tx_oidc[code]=' . $_GET['code'];
-        if (!empty($_SESSION['oidc_redirect_url']) && strpos($loginUrl, 'redirect_url=') === false) {
-            $loginUrl .= '&redirect_url=' . urlencode($_SESSION['oidc_redirect_url']);
+        $loginUrl .= false !== strpos($loginUrl, '?') ? '&' : '?';
```

Comment:
Just for the info, I'm against yoda-style code, so everything of this form is a no-go for me.